src/sources/closed_coil.cpp

```diff
@@ -0,0 +1,634 @@
+#include "closed_coil.hpp"
+
+namespace hephaestus {
+
+// Pushes an element into a vector if the vector does not yet contain that same
+// element
+template <typename T> void pushIfUnique(std::vector<T> &vec, const T el) {
+
+  bool verify = true;
+
+  for (auto e : vec) {
+    if (e == el)
+      verify = false;
+  }
+
+  if (verify == true)
+    vec.push_back(el);
+}
+
+// Deletes and clears a vector of pointers
+template <typename T> void deleteAndClear(std::vector<T *> v) {
+
+  for (auto p : v)
+    delete p;
+  v.clear();
+}
+
+double highV(const mfem::Vector &x, double t) { return 1.0; }
+double lowV(const mfem::Vector &x, double t) { return 0.0; }
+
+// Base class methods
+
+ClosedCoilSolver::ClosedCoilSolver(
+    const hephaestus::InputParameters &params,
+    const std::vector<hephaestus::Subdomain> &coil_dom,
+    const int electrode_face, const int order)
+    : hcurl_fespace_name_(params.GetParam<std::string>("HCurlFESpaceName")),
+      J_gf_name_(params.GetParam<std::string>("JGridFunctionName")),
+      I_coef_name_(params.GetParam<std::string>("IFuncCoefName")),
+      coil_domains_(coil_dom), order_(order), coef1_(nullptr), coef0_(nullptr),
+      mesh_parent_(nullptr), J_parent_(nullptr), HCurlFESpace_parent_(nullptr) {
+
+  elec_attrs_.first = electrode_face;
+}
+
+ClosedCoilSolver::~ClosedCoilSolver() {
+
+  delete coef1_;
+  delete coef0_;
+  delete high_src_;
+  delete low_src_;
+
+  deleteAndClear(mesh_);
+  deleteAndClear(H1FESpace_);
+  deleteAndClear(HCurlFESpace_);
+  deleteAndClear(H1_Collection_);
+  deleteAndClear(HCurl_Collection_);
+  deleteAndClear(V_);
+  deleteAndClear(sps_);
+  deleteAndClear(sps_params_);
+  deleteAndClear(current_solver_options_);
+  deleteAndClear(gridfunctions_);
+  deleteAndClear(fespaces_);
+  deleteAndClear(bc_maps_);
+  deleteAndClear(coefs_);
+  deleteAndClear(high_DBC_);
+  deleteAndClear(low_DBC_);
+}
+
+void ClosedCoilSolver::Init(hephaestus::GridFunctions &gridfunctions,
+                            const hephaestus::FESpaces &fespaces,
+                            hephaestus::BCMap &bc_map,
+                            hephaestus::Coefficients &coefficients) {
+
+  coef1_ = new mfem::ConstantCoefficient(1.0);
+  coef0_ = new mfem::ConstantCoefficient(0.0);
+
+  // Retrieving the parent FE space and mesh
+  HCurlFESpace_parent_ = fespaces.Get(hcurl_fespace_name_);
+  if (HCurlFESpace_parent_ == NULL) {
+    const std::string error_message = hcurl_fespace_name_ +
+                                      " not found in fespaces when "
+                                      "creating ClosedCoilSolver\n";
+    mfem::mfem_error(error_message.c_str());
```

Comment:
No particular reason, I'm just following the way it was done in the DivFreeSource class. I could change all of these for MFEM_ABORTs and ASSERTs if that's better.

Reply:
Hephaestus is indeed inconsistent between use of `MFEM_ABORT` and `mfem::mfem_error` elsewhere - probably worth a separate PR to address this, to avoid making this one too sprawling!